platform/core/src/services/HangingProtocolService/HangingProtocolService.ts

```diff
@@ -313,7 +297,20 @@ class HangingProtocolService {
     return protocol;
   }
 
-  _setProtocol(protocol) {
+  public applyProtocol(
+    protocol: HangingProtocol.Protocol,
+    matchingDisplaySets: Record<string, HangingProtocol.DisplaySetMatchDetails>
```

Comment:
It should be allowed to supply the matchingDisplaySets optionally - it is entirely possible to just reapply a specific protocol without specifying the display sets